Source/gui/SNTBinaryMessageWindowView.swift

```diff
@@ -269,41 +273,51 @@ struct SNTBinaryMessageWindowView: View {
     ) {
       SNTBinaryMessageEventView(e: event!, customURL: customURL, bundleProgress: bundleProgress)
 
-      VStack(spacing: 15.0) {
-        SNTNotificationSilenceView(
-          silence: $preventFutureNotifications,
-          period: $preventFutureNotificationPeriod
-        )
+      SNTNotificationSilenceView(silence: $preventFutureNotifications, period: $preventFutureNotificationPeriod)
 
-        if event?.needsBundleHash ?? false && !bundleProgress.isFinished {
-          if bundleProgress.fractionCompleted == 0.0 {
-            ProgressView {
-              Text(bundleProgress.label)
-            }.progressViewStyle(.linear)
-          } else {
-            ProgressView(value: bundleProgress.fractionCompleted) {
-              Text(bundleProgress.label)
-            }
+      if event?.needsBundleHash ?? false && !bundleProgress.isFinished {
+        if bundleProgress.fractionCompleted == 0.0 {
+          ProgressView() {
+            Text(bundleProgress.label)
+          }.progressViewStyle(.linear)
+        } else {
+          ProgressView(value: bundleProgress.fractionCompleted) {
+            Text(bundleProgress.label)
           }
         }
+      }
 
-        HStack(spacing: 15.0) {
-          if c.eventDetailURL != nil
-            && !(event?.needsBundleHash ?? false && !bundleProgress.isFinished)
-          {
-            OpenEventButton(customText: c.eventDetailText, action: openButton)
-          }
-          DismissButton(
-            customText: c.dismissText,
-            silence: preventFutureNotifications,
-            action: dismissButton
-          )
+      // Display the standalone error message to the user if one is provided.
+      if let errorMessage = standaloneErrorMessage {
```

Comment:
consider using the same logic as for displaying the button here (calling `addStandaloneButton`) instead of the `if let` statement. Might be good to change the method name too so it's less generic.

Reply:
Fair the `if let` was to handle the optional string as the `if let` keyword unboxes it from the optional same as Rust. 

I can wrap it in in a `addStandaloneButton` check

Reply:
Wrapped this in a simple check

```swift
      // Display the standalone error message to the user if one is provided.
      if c.enableStandaloneMode {
        if let errorMessage = standaloneErrorMessage {
          if errorMessage != "" {
            Text(errorMessage).foregroundColor(.red)
          }
        }
      }
```